dvc/output.py

```diff
@@ -514,6 +515,23 @@ def index_key(self) -> Tuple[str, "DataIndexKey"]:
             key = self.fs.path.parts(no_drive)[1:]
         return workspace, key
 
+    def get_entry(self) -> "DataIndexEntry":
```

Comment:
Wondering if this should be backwards: data index is getting filled up first and then output gets info from it. SInce we want to work directly with index during operations (e.g. add/checkout/etc), it means that it should be the source of truth that is used during serializing into dvcfiles. Obviously we could try to propagate the changes back into outputs (like we do right now in cloud versioning), but I wonder if that's actually a wrong way to approach it. WDYT?